app/org/sagebionetworks/bridge/validators/ActivityValidator.java

```diff
@@ -47,10 +57,12 @@ private void validate(Errors errors, TaskReference ref) {
         errors.pushNestedPath("task");
         if (isBlank(ref.getIdentifier())) {
             errors.rejectValue("identifier", CANNOT_BE_BLANK);
+        } else if (taskIdentifiers == null || !taskIdentifiers.contains(ref.getIdentifier())) {
+            errors.rejectValue("identifier", geTaskIdentifierMessage(ref));
```

Comment:
Minor: Would it be cleaner to have the null check in the constructor, and then substitute it with an empty set? Functionality should be the same, but I feel like it would be cleaner since it moves the null check closer to the source.